File: solutions/uri/2152/2152.cpp
#include <iostream>

int main() {
    int test_cases, h, m, o;

    std::cin >> test_cases;

    while (test_cases--) {
        std::cin >> h >> m >> o;
        if (h < 10) {
            std::cout << "0" << h;
        } else {
            std::cout << h;
        }
        std::cout << ":";
        if (m < 10) {
            std::cout << "0" << m;
        } else {
            std::cout << m;
        }
        std::cout << " - A porta ";
        if (o == 1) {
            std::cout << "abriu!";
        } else {
            std::cout << "fechou!";
        }
        std::cout << std::endl;
    }

    return 0;
}

File: solutions/beecrowd/2152/2152.py
import sys

input = sys.stdin.readline
output = sys.stdout.write

test_cases = int(input())

for _ in range(test_cases):
    h, m, o = map(int, input().split())
    output(f'{h:02d}:{m:02d} - A porta {"abriu" if o == 1 else "fechou"}!\n')
